eos-update-flatpak-repos

```diff
@@ -635,9 +635,29 @@ def _migrate_installed_flatpaks():
             try:
                 new_branch = migrate.get('new-branch')
                 if new_branch:
-                    _update_branch(ref, new_branch, refs)
-                    branch = new_branch
-                    ref = _refresh_ref(inst, kind, name, arch, branch)
+                   new_ref = None
+                   try:
+                       new_ref = inst.get_installed_ref(kind, name, arch,
+                                                        new_branch)
+                   except GLib.GError as e:
+                       if e.matches(Flatpak.error_quark(),
+                                    Flatpak.Error.NOT_INSTALLED):
+                           pass
+                       else:
+                           raise
+
+                   if new_ref:
+                       logging.info('{} already installed, removing {}'
+                                    .format(new_ref.format_ref(),
+                                            ref.format_ref()))
+                       inst.uninstall_full(Flatpak.UninstallFlags.NO_PRUNE,
+                                           kind, name, arch, branch)
```

Comment:
Why do we continue on into the code that updates the deploy file in this case? I don't think it's quite right because `new_ref`'s origin may in theory be neither `old-origin` nor `new-origin`.